File: backend/internal/models/message.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Message struct {
	// gorm.Model
	ID             uuid.UUID `gorm:"unique;type:uuid;primaryKey"`
	Content        string    `gorm:"not null"`
	Readed         bool      `gorm:"not null"`
	AuthorId       uuid.UUID
	Author         User `gorm:"foreignKey:AuthorId"`
	SentAt         time.Time
	ConversationId uuid.UUID
	Conversation   Conversation `gorm:"foreignKey:ConversationId"`
	CreatedAt      time.Time
	UpdatedAt      time.Time
	DeletedAt      *time.Time `gorm:"index"`
}

File: frontend/lib/core/models/message.dart
class Message {
  final String id;
  final String author;
  final String authorId;
  final String content;
  final DateTime createdAt;
  final DateTime updatedAt;
  final bool readed;

  Message({
    required this.id,
    required this.author,
    required this.authorId,
    required this.content,
    required this.createdAt,
    required this.updatedAt,
    required this.readed
  });

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      id: json['ID'],
      author: json['Author'],
      authorId: json['AuthorID'],
      content: json['Content'],
      createdAt: DateTime.parse(json['CreatedAt']),
      updatedAt: DateTime.parse(json['UpdatedAt']),
      readed: json['Readed']
    );
  }
}